### PR TITLE
add fast support and option

### PR DESCRIPTION
**Pull request by ArthurZucker**
# What does this PR do?
Adresses #22669

### CODE REVIEW

File: src/transformers/models/llama/convert_llama_weights_to_hf.py [hunk 1]
```diff
@@ -17,12 +17,21 @@
...
 
 from transformers import LlamaConfig, LlamaForCausalLM, LlamaTokenizer
 
 
+try:
+    from transformers import LlamaTokenizerFast
+except ValueError as e:
+    warnings.warn(e)
+    warnings.warn(
+        "The converted tokenizer will be the `slow` tokenizer. To use the fast, update your `tokenizer` library and re-run the tokenizer conversion"
```
Comment (sgugger) [inline src/transformers/models/llama/convert_llama_weights_to_hf.py:32]:
```suggestion
        "The converted tokenizer will be the `slow` tokenizer. To use the fast, update your `tokenizers` library and re-run the tokenizer conversion"
```

File: src/transformers/models/llama/convert_llama_weights_to_hf.py [hunk 2]
```diff
@@ -17,12 +17,21 @@
...
+    )
+
```
Comment (sgugger) [inline src/transformers/models/llama/convert_llama_weights_to_hf.py:34]: Set `LlamaTokenizerFast` to None there.

File: src/transformers/models/llama/convert_llama_weights_to_hf.py [hunk 3]
```diff
@@ -231,10 +240,16 @@ def permute(w):
     shutil.rmtree(tmp_model_path)
 
 
-def write_tokenizer(tokenizer_path, input_tokenizer_path):
+def write_tokenizer(tokenizer_path, input_tokenizer_path, use_fast=True):
```
Comment (sgugger) [inline src/transformers/models/llama/convert_llama_weights_to_hf.py:243]:
```suggestion
def write_tokenizer(tokenizer_path, input_tokenizer_path):
```
Let's just do both in the conversion script. It should create a complete repo.

File: src/transformers/models/llama/convert_llama_weights_to_hf.py [hunk 4]
```diff
@@ -231,10 +240,16 @@ def permute(w):
...
     print(f"Fetching the tokenizer from {input_tokenizer_path}.")
     # Initialize the tokenizer based on the `spm` model
-    tokenizer = LlamaTokenizer(input_tokenizer_path)
+    tokenizer_class = LlamaTokenizer
+    if use_fast:
+        try:
+            tokenizer_class = LlamaTokenizerFast
+        except ValueError:
+            pass
```
Comment (sgugger) [inline src/transformers/models/llama/convert_llama_weights_to_hf.py:251]:
```suggestion
    tokenizer_class = LlamaTokenizer if LlamaTokenizerFast is None else LlamaTokenizerFast
```

File: src/transformers/models/llama/convert_llama_weights_to_hf.py [hunk 5]
```diff
@@ -252,6 +267,11 @@ def main():
         "--output_dir",
         help="Location to write HF model and tokenizer",
     )
+    parser.add_argument(
+        "--use_fast_tokenizer",
+        default=True,
+        help="Location to write HF model and tokenizer",
+    )
```
Comment (sgugger) [inline src/transformers/models/llama/convert_llama_weights_to_hf.py:274]: No need to add an argument.

**Review by sgugger — COMMENTED** [review]
Thanks for working on this, added a couple of comments!

File: src/transformers/models/llama/convert_llama_weights_to_hf.py (hunk 4 above)
Comment (amyeroberts) [inline src/transformers/models/llama/convert_llama_weights_to_hf.py:251]: I'd print a message here saying `use_fast` is `True` but the fast tokenizer couldn't be loaded, as the other warning will happen when CLI is first called, so might not be spotted. This is just a weak suggestion though :)

**Review by amyeroberts — APPROVED** [review]
Very nice. Thanks for updating this!

File: src/transformers/models/llama/convert_llama_weights_to_hf.py [hunk 6]
```diff
@@ -17,12 +17,21 @@
...
+import warnings
 
 import torch
 
 from transformers import LlamaConfig, LlamaForCausalLM, LlamaTokenizer
 
 
+try:
+    from transformers import LlamaTokenizerFast
+except ValueError as e:
```
[the remark this answers is not in the document]
Comment (ArthurZucker) [reply src/transformers/models/llama/convert_llama_weights_to_hf.py]: Indeed value errors are in the checking

**Review by sgugger — APPROVED** [review]

**Pull request merged**